test/integration/orm_action_execution.go

```diff
@@ -0,0 +1,216 @@
+	namespacescope_operand_container_name  = "icp-audit-service"
+	namespacescope_operator_cr_name        = "example-authentication"
+	namespacescope_operand_path_in_cr      = ".spec.auditService.resources"
+
+	clusterscope_operand_deployment_name = "cert-manager-cainjector"
+	clusterscope_operand_container_name  = "cert-manager-cainjector"
+	clusterscope_operator_cr_name        = "default"
+	clusterscope_operand_path_in_cr      = ".spec.certManagerCAInjector.resources"
+)
+
+var (
+	groupVersionRes4nsscope = schema.GroupVersionResource{
+		Group:    "operator.ibm.com",
+		Version:  "v1alpha1",
+		Resource: "authentications",
+	}
+	groupVersionRes4clusterscope = schema.GroupVersionResource{
+		Group:    "operator.ibm.com",
+		Version:  "v1alpha1",
+		Resource: "certmanagers",
+	}
+)
+
+var _ = Describe("Action Executor ", func() {
+	f := framework.NewTestFramework("orm-action-executor")
+	var kubeConfig *restclient.Config
+	var actionHandler *action.ActionHandler
+	var kubeClient *kubeclientset.Clientset
+	var dynamicClient dynamic.Interface
+	var origCRValue4NsScope, origCRValue4ClusterScope interface{}
+	var needRevertCR4NsScope, needRevertCR4ClusterScope bool
+
+	BeforeEach(func() {
+		f.BeforeEach()
+		// The following setup is shared across tests here
+		if kubeConfig == nil {
+
+			kubeConfig := f.GetKubeConfig()
+			kubeClient = f.GetKubeClient("orm-action-executor")
+
+			var err error
+			dynamicClient, err = dynamic.NewForConfig(kubeConfig)
+			if err != nil {
+				framework.Failf("Failed to generate dynamic client for kubernetes test cluster: %v", err)
+			}
+
+			s := app.NewVMTServer()
+			kubeletClient := s.CreateKubeletClientOrDie(kubeConfig, kubeClient, "", "busybox", map[string]set.Set{}, true)
+			apiExtClient, err := apiextclient.NewForConfig(kubeConfig)
+			if err != nil {
+				glog.Fatalf("Failed to generate apiExtensions client for kubernetes target: %v", err)
+			}
+			ormClient := resourcemapping.NewORMClient(dynamicClient, apiExtClient)
+
+			probeConfig := createProbeConfigOrDie(kubeClient, kubeletClient, dynamicClient)
+
+			discoveryClientConfig := discovery.NewDiscoveryConfig(probeConfig, nil, app.DefaultValidationWorkers,
+				app.DefaultValidationTimeout, aggregation.DefaultContainerUtilizationDataAggStrategy,
+				aggregation.DefaultContainerUsageDataAggStrategy, ormClient, app.DefaultDiscoveryWorkers, app.DefaultDiscoveryTimeoutSec,
+				app.DefaultDiscoverySamples, app.DefaultDiscoverySampleIntervalSec)
+			actionHandlerConfig := action.NewActionHandlerConfig("", nil, nil,
+				cluster.NewClusterScraper(kubeClient, dynamicClient, false, nil, ""),
+				[]string{"*"}, ormClient, false, true, 60, gitops.GitConfig{})
+			actionHandler = action.NewActionHandler(actionHandlerConfig)
+
+			// Kubernetes Probe Discovery Client
+			discoveryClient := discovery.NewK8sDiscoveryClient(discoveryClientConfig)
+			_, _, err = discoveryClient.DiscoverWithNewFramework("discovery-integration-test")
```

Comment:
Why do we need this here? Wouldn't this slow the whole test down a lot, especially when its against a real cluster

Reply:
I have to call `NewK8sDiscoveryClient` to trigger `CacheORMSpecMap` to initialize the map for saving orm object. Probably I could call it directly `dc.config.ormClient.CacheORMSpecMap()`, not sure if it works or not, will give it a shot.

Reply:
@irfanurrehman , I can't call this function like this `dc.config.ormClient.CacheORMSpecMap()` , `config` is not visiable outside the package. so I have to keep the old way.

Reply:
I am concerned about the time wasted in discovery on this cluster. I am not sure how much time does it take on this cluster. Can you please confirm. If its lesser then a minute, I am ok with it.

Reply:
@irfanurrehman , I added a function named `TimeCost` to calculate the time consumption of `DiscoverWithNewFramework`. In the Travis log, we could see the following output, it is less than 20s, I also test this on my local Kind cluster, it's around 15s. 
```
Function name<DiscoverWithNewFramework>,Run time<7.955075221s>
Function name<DiscoverWithNewFramework>,Run time<15.17894403s>
```

Reply:
Did you try exporting the variables and make `dc.config.ormClient.CacheORMSpecMap()` usable here?
My point is that we have to be cautious in adding time to overall test execution run. I think its already topping 25 mins now. Every 20 seconds add to it and there still are more tests to come.

Reply:
I tried it out, if we export the variables, we call `dc.config.ormClient.CacheORMSpecMap()` directly, there is a significant improvement, you can check that in Travis log.
```
Function name<DiscoverWithNewFramework>,Run time<816.624924ms>
Function name<DiscoverWithNewFramework>,Run time<775.385891ms>
```